src/config.h

```diff
@@ -5,3 +5,23 @@
 #else
 	#warning Missing configuration
 #endif
+
+/*
+ * This is here because CMake won't let us a define a function-style macro from CMakeLists.
+ * And we're using a function-style macro so that using it undefined raises an error,
+ * because '#if FOO' or '#if FOO == 1' is silently false (mandated by C standard).
+ *
+ * in short:
+ * - IS_CLIENT_BUILD is for optional helpers in header files and similar
+ * - CHECK_CLIENT_BUILD() is for cases where program behavior differs
```

Comment:
I think it's more important to mention the problem that the macros attempt to address first. Then followed by an explanation why this is implemented as-is. For some reason it took me 5 minutes to realize that `CHECK_CLIENT_BUILD` is not defined on server builds, thus throwing a fancier error.


```suggestion
 * CHECK_CLIENT_BUILD() : use in function code (usually in source files). This macro is not defined
 *                        for server builds, resulting in a more useful compile-time error.
 * IS_CLIENT_BUILD : use in client/server shared files to skip unneeded (function) declarations
 *
 * The underlying problem: '#if FOO' or '#if FOO == 1' is silently false (mandated by C standard),
 * thus we're using a function-style macro. Unfortunately such cannot be defined from CMakeLists.
```

Reply:
looks like I need to explain it better, because `CHECK_CLIENT_BUILD()` is defined in server builds (it's just 0).
it is not defined when compiling objects that are shared between client and server